mm2src/mm2_event_stream/src/event.rs

```diff
@@ -0,0 +1,45 @@
+use serde_json::Value as Json;
+
+// Note `Event` shouldn't be `Clone`able, but rather Arc/Rc wrapped and then shared.
+// This is only for testing.
+#[cfg_attr(test, derive(Clone, Debug, PartialEq))]
+#[derive(Default)]
+/// Multi-purpose/generic event type that can easily be used over the event streaming
+pub struct Event {
+    /// The type of the event (balance, network, swap, etc...).
+    event_type: String,
+    /// The message to be sent to the client.
+    message: Json,
+    /// Indicating whether this event is an error event or a normal one.
+    error: bool,
+}
+
+impl Event {
+    /// Creates a new `Event` instance with the specified event type and message.
+    #[inline]
+    pub fn new(streamer_id: String, message: Json) -> Self {
+        Self {
+            event_type: streamer_id,
+            message,
+            error: false,
+        }
+    }
+
+    /// Create a new error `Event` instance with the specified error event type and message.
+    #[inline]
+    pub fn err(streamer_id: String, message: Json) -> Self {
+        Self {
+            event_type: streamer_id,
+            message,
+            error: true,
+        }
+    }
+
+    /// Returns the `event_type` (the ID of the streamer firing this event).
+    pub fn origin(&self) -> &str { &self.event_type }
+
+    pub fn get(&self) -> (String, Json) {
+        let prefix = if self.error { "ERROR:" } else { "" };
+        (format!("{prefix}{}", self.event_type), self.message.clone())
+    }
```

Comment:
This doesn't look very clear at the first glance. Can you document what this is?

Reply:
[547fc2b](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2172/commits/547fc2bb4c03980379cfd219995e1d5ffa199bde)